torch/distributed/tensor/parallel/_utils.py

```diff
@@ -46,15 +48,53 @@ def _validate_tp_mesh_dim(
 
     parent_mesh = _mesh_resources.get_parent_mesh(device_mesh)
     if parent_mesh:
-        if parent_mesh.ndim != 2:
-            raise RuntimeError(
-                f"Found TP device_mesh has a parent mesh with dims {parent_mesh.ndim}",
-                "Currently we only support 2D TP composition with DP.",
-            )
+        # if parent_mesh.ndim != 2:
+        #     raise RuntimeError(
+        #         f"Found TP device_mesh has a parent mesh with dims {parent_mesh.ndim}",
+        #         "Currently we only support 2D TP composition with DP.",
+        #     )
 
         tp_mesh_dim = _mesh_resources.get_parent_mesh_dim(device_mesh)
-        if tp_mesh_dim != 1:
-            raise RuntimeError(
-                f"Found TP device_mesh on the {tp_mesh_dim} dimension of its parent mesh.",
-                "Currently we only support intranode TP and TP needs to be the innermost dimension on its parent mesh.",
-            )
+        # if tp_mesh_dim != 1:
```

Comment:
can we preserve this to be sth like `if tp_mesh_dim != parent_mesh.ndim -1`?

Reply:
oops almost missed this part